src/annoylib.h

```diff
@@ -233,10 +243,11 @@ template<typename S, typename T, template<typename, typename, typename> class Di
    * We create a tree like this q times. The default q is determined automatically
    * in such a way that we at most use 2x as much memory as the vectors take.
    */
-protected:
-  typedef Distance<S, T, Random> D;
-  typedef typename D::Node Node;
+public:
+  typedef Distance D;
+  typedef typename D::template Node<S, T> Node;
```

Comment:
It's somewhat like when you have to use typename on nested types.  It just tells the compiler that D::Node is a template (and not a variable, standard type, etc.)